### PR TITLE
Refactor Spark Connector into two modules for reusability

### PR DESCRIPTION
**Pull request by cbalci**
This is the first of a two PR changes which will add Spark3 support for 'pinot-spark-connector'

**Background**
Apache Spark has [changed](https://blog.madhukaraphatak.com/spark-3-datasource-v2-part-3) the Datasource interface significantly between Spark2 and Spark3 so `pinot-spark-connector` doesn't work for Spark3. We can implement a new connector for Spark3 as a separate module, however about half of the logic/code under the existing connector is independent of the interface and can potentially be reused across Spark2 and Spark3 connectors. For this, I'm restructuring the packages similar to what was done for batch ingestion in https://github.com/apache/pinot/pull/8560.

**Change**
In this PR, I'm refactoring Spark Connector into two packages as:
`pinot-spark-connector` --> ( `pinot-spark-common` +  `pinot-spark-2-connector` )

This is mostly a mechanical refactoring which moves packages around and renames fields/classes for clarity. Only addition is the backported (from Spark) `CaseInsensitiveStringMap` to make `PinotDataSourceReadOptions` reusable across implementations (see comment below).

**Testing**
Usage and functionality of the Spark2 connector should be completely unchanged except for the renaming of the maven module. All the unit tests are preseved to ensure previous assumptions. I also ran the integration tests under `ExampleSparkPinotConnectorTest` to verify expected behavior.

To preview the full changes including the Spark3 Connector implementation you can check [this diff](https://github.com/apache/pinot/compare/master...cbalci:pinot:pinot-spark3-connector#diff-8733e0d7481c08afa1005dfcebcf93aae07cfdc996c82f809bef22fcb2061e0eR41).

`refactor` `cleanup`
`release-notes` ('Pinot Spark Connector' module is renamed to 'Pinot Spark 2 Connector' for clarity)

### CODE REVIEW

**Review by xiangfu0 — APPROVED** [review]
LGTM

File: pinot-connectors/pinot-spark-common/src/main/java/org/apache/pinot/connector/spark/common/CaseInsensitiveStringMap.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
+/**
```
[the remark this answers is not in the document]
Comment (xiangfu0) [reply pinot-connectors/pinot-spark-common/src/main/java/org/apache/pinot/connector/spark/common/CaseInsensitiveStringMap.java:1]: how about make this an interface and have the wrapper of v2/v3 implementation ?

[the remark this answers is not in the document]
Comment (cbalci) [reply pinot-connectors/pinot-spark-common/src/main/java/org/apache/pinot/connector/spark/common/CaseInsensitiveStringMap.java:1]: I think this class enables the simplest interface we can expose to the implementor: `Map[String,String]`. 
If you take a look at the `PinotDataSourceReadOptions` below, it has two factory methods with following signatures:
```
object PinotDataSourceReadOptions {
...
  private[pinot] def from(optionsMap: util.Map[String, String]): PinotDataSourceReadOptions
  ...
  private[pinot] def from(options: CaseInsensitiveStringMap): PinotDataSourceReadOptions
...
}
```
With this, the shared `PinotDataSourceOptions` object can be created either by passing a `CaseInsensitiveStringMap` or a plain old `Map` which internally will be converted to a `CaseInsensitiveStringMap`. I guess we can even drop the second method and only accept `Map[String, String]` for simplicity.

Let me know if you meant something else.

**Comment by KKcorps** [comment]
Will take a look at it today!

**Comment by KKcorps** [comment]
@cbalci Should I go ahead and merge this?

**Comment by cbalci** [comment]
Thanks for the review @KKcorps ! Yes, please go ahead. I have a couple follow up PRs waiting for this.

**Pull request merged**